### PR TITLE
Incorrect placement of  update submodules on checkout  setting #7290

### LINKED ISSUE #7290: Incorrect placement of "Update submodules on checkout" setting
## Current behaviour

It is possible to configure whether submodules are updated on checkout (e.g. when a revision is checkout).
As discovered as part of analysis for #7274, the functionality is guarded by the following checkbox:
grafik
However it is totally unexpected and confusing to a user, as the page is about configuring confirmations, and not underlying settings.

## Expected behaviour

* The existing checkbox is moved to another settings page. Proposed placement "General" page under "Performance" group, after "Show submodules status in browser menu" item.
* Add a new `DontConfirmUpdateSubmodulesOnCheckout` setting that will guard against showing confirmations
* Update all code paths to correctly use these settings

### PR DESCRIPTION
**Pull request by manishkungwani**
Fixes #7290
- Moved the "Update submodules on checkout" setting from Confirmation page to General page (Behaviour group).

## Proposed changes

- [x] Added the control to General page
- [x] Added the logic to General page
- [x] Removed the control to Confirmations page
- [x] Removed the logic from Confirmations page
- [x] Ensured that TabOrder is correct

## Screenshots

image

image

## Test methodology 

- Verified the UI
- Tested the settings are not adversely affected by the change in UI.

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

**Comment by gerhardol** [comment]
As seen on the build page:
https://ci.appveyor.com/project/gitextensions/gitextensions/builds/28101391
Please update English translations and re-submit the pull-request.
For more information please refer to https://github.com/gitextensions/gitextensions/wiki/Translations

Is not the confirmation for the dialog still wanted?
(the settings will be split now, hard to explain that the Behaviour and Confirmation settings are related).

**Comment by manishkungwani** [comment]
@gerhardol I did update the translations, but it remained the same. This happened, since the control IDs did not change and the translations remained the same as well.

Here is the output of updating translations:

image

**Comment by RussKie** [comment]
Thank you, though few things need further work
 
* We need to add a new `DontConfirmUpdateSubmodulesOnCheckout` setting that will guard against showing confirmations.
The setting will reside in the original placement of "Update submodules on checkout". So in essence, we need to rollback the change to ConfirmationsSettingsPage.cs, rename the checkbox and use the new settings.
* This new setting needs to be respected when showing confirmation box.

Please let me know if you have questions.

**Comment by manishkungwani** [comment]
@RussKie the current setting also does a similar function, so pl confirm if this is correct:
- **UpdateSubmodulesOnCheckout** - Its state will define whether the Submodules are updated on Checkout
- If _True_, Submodules will be updated on Checkout
- If _False_, upon Checkout, we shall check the new setting **DontConfirmUpdateSubmodulesOnCheckout** and show a confirmation, if **DontConfirmUpdateSubmodulesOnCheckout** is _True_
- If **DontConfirmUpdateSubmodulesOnCheckout** is _False_, the Checkout process won't be interrupted

**Comment by RussKie** [comment]
Yes, that is correct - 2 settings: one to toggle the functionality, and another to toggle confirmations.
Thanks

**Comment by gerhardol** [comment]
>  I hope this clarifies the confirmation comment from you as well.

It should be OK now, I have not looked at the implementation

**Comment by manishkungwani** [comment]
@RussKie pl review.

File: GitUI/CommandsDialogs/SettingsDialog/Pages/ConfirmationsSettingsPage.Designer.cs [hunk 1]
```diff
@@ -31,13 +31,6 @@ private void InitializeComponent()
             this.tlpnlMain = new System.Windows.Forms.TableLayoutPanel();
             this.gbConfirmations = new System.Windows.Forms.GroupBox();
             this.tlpnlConfirmations = new System.Windows.Forms.TableLayoutPanel();
-            this.chkFetchAndPruneAllConfirmation = new System.Windows.Forms.CheckBox();
```
Comment (RussKie) [inline GitUI/CommandsDialogs/SettingsDialog/Pages/ConfirmationsSettingsPage.Designer.cs]: Could you please clarify what changed here?

File: GitUI/MessageBoxes.cs [hunk 2]
```diff
@@ -89,6 +89,7 @@ public static bool ConfirmUpdateSubmodules(IWin32Window win)
 
             if (PSTaskDialog.cTaskDialog.VerificationChecked)
             {
+                AppSettings.DontConfirmUpdateSubmodulesOnCheckout = result;
```
Comment (RussKie) [inline GitUI/MessageBoxes.cs]: Why are these two updated to the same value?

**Comment by manishkungwani** [comment]
@RussKie do you need anything else from me on this?

**Comment by manishkungwani** [comment]
@RussKie can you pl confirm if this needs any more work.

Thank you

**Comment by RussKie** [comment]
Please resolve the MC and squash.

**Comment by RussKie** [comment]
@msftbot merge in 12 hours

**Pull request merged**